### DISCUSSION TITLE
Documentation Generator Error

### DISCUSSION THREAD

**Discussion by sgp-infainite** [pragunbhutani/dbt-llm-agent#39] [Q&A]
Great utility you are building here, loving the chat function.

Now looking to explore the Documentation Generator and hitting a path error.

For simplicity using the example_dbt_project.

Can successfully set the dbt_project_root="example_dbt_project", however the model name seems to be throwing the error below. path/model name issue potentially?

<img width="288" alt="Screenshot 2024-07-12 at 1 33 15 PM" src="https://github.com/user-attachments/assets/aed79073-e746-4a38-943a-da66740b7f5c">
<img width="736" alt="Screenshot 2024-07-12 at 1 33 05 PM" src="https://github.com/user-attachments/assets/57829419-4373-4395-8d0e-5c3ed209c7f9">

(venv) homedir@compname dbt_llm_tools % python document_generator.py
Traceback (most recent call last):
File "/Users/homedir/Documents/repos/dbt-llm-tools/dbt_llm_tools/document_generator.py", line 10, in
doc_gen.generate_documentation(
File "/Users/homedir/Documents/repos/dbt-llm-tools/dbt_llm_tools/documentation_generator.py", line 201, in generate_documentation
for dep in model.get("deps", []):
^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'

**Comment by pragunbhutani** [comment]
Hey, thanks for pointing that out. You're right, it seems like something went wrong with the project parser and it hasn't been able to parse the model successfully. I suspect it's to do with the way we handle the path variable. I'll try to recreate the error and see if I can fix it.

**Comment by pragunbhutani** [comment]
The documentation generator is no longer explicitly callable from the backend, but you can access it through the UI.
